### ISSUE TITLE
Provide an official way for scripts to provide script-wide help

### ISSUE THREAD

**Issue by oakkitten** [weechat/weechat#1021]
Rationale:

- currently many scripts don't provide help for the script in the form of `/help script_name`. it might be that some script authors feel that creating a bogus command just for help would be too much of a workaround.
- some scripts that do have the command `/script_name` only provide help for that command. it is often not clear if what `/help script_name` prints is *everything* that script does, or if only related to that command.
- the only way to see a bit more about a script in `/script` is to look at the source of the script. again, not all scripts have help there, and those that do may include formatting, licensing information or otherwise may be hard to read. besides, it is not obvious that there's help in “view script”. if there was a standardized way of providing help, there could be a command “view script manual” or something

I suggest `/help script_name.extension` or `/script info script_name.extension`

**Comment by sim642** [comment]
This is especially bad for scripts where the script name is different from the script's command, like `buffer_autosetpy` has `/autosetbuffer`, and there is nothing preventing scripts from doing this, which makes it hard to find the relevant help even, so I am in favor of this feature.

I think `/help script_name` syntax (without the extension) works just as well because the script plugin can work with loaded scripts without needing to specify the extension. If the script and the command have the same name, both's helps could be printed, probably the script's one first would make more sense. For scripts which are not loaded but exist in the scripts repo, some command under `/script` should probably exist to show the same information.

For loaded scripts the help could also contain a list of commands the script has hooked and maybe something else similar. This would make it easy to find which command(s) the script provides if the help doesn't, especially for scripts which are not updated to have such script-wide help.

**Comment by flashcode** [comment]
I think the current way to provide help via a command having the same name as the script is pretty good.

All scripts could provide help in this way (not only on the command, but on the script itself).
@sim642 I agree that buffer_autoset.py (which I wrote) should have a command with the same name as the script, so I will fix that and rename the command.

Maybe this is not needed to add a new way to provide help because:

1. that would affect all existing scripts (and it will take lot of time to update all existing scripts to provide help in this way)
2. this could be duplicated with the help given by the command.

Instead, I can be a bit more restrictive on the upload of new scripts by requiring:

1. a command, with the same name as the script
2. the command must explain how the script works (if needed), and what does the command (2 paragraphs are OK in the help).

And @sim642 suggested a way to show help on a script in repo (which is not installed), it's not easy to do that until you load the script in WeeChat: how would you parse the help from the script, in the 7 (and soon 8) supported languages?

**Comment by weechatter** [comment]
In my opinion its already possible:

/help <scriptname>
can provide an "script-wide" help for the script.

If the script supports additional commands its also possible to add help for every single command. But at least the script author has to implement it. A new "script-wide" help will not solve the problem of "not writing a help text". Also the short description is enough, because we are talking about some KB to download for a single script and people don't even read the short description ;-)

**Comment by tomoe-mami** [comment]
> ... upload of new scripts by requiring:
> 1.  a command, with the same name as the script


@flashcode but what if the script absolutely doesn't need a command or help text (for example, [here's one][1] that I haven't submitted). The short description from `weechat.register` already explains what the script does. Won't requiring a command for that script just a waste of hook?


[1]: https://github.com/tomoe-mami/weechat-scripts/tree/master/empty_complete

**Comment by flashcode** [comment]
@tomoe-mami: yes maybe not appropriate for all scripts. But then there could be a `/script xxx` command that could show all the commands / options defined by the script (this is already displayed when you do "v" + Enter on a script in /script buffer).

**Comment by oakkitten** [comment]
> Instead, I can be a bit more restrictive on the upload of new scripts by requiring

i have written a script with a command that is script_name. the description of the command is ~10 lines, script help is ~80 lines. i don't want to lump everything together. still, this would better the current situation

>  it's not easy to do that until you load the script in WeeChat

how do you currently get the version and the short description? can't you get the long description in the same way?

> A new "script-wide" help will not solve the problem of "not writing a help text".

maybe not by itself, but making it an official requirement will

**Comment by flashcode** [comment]
@oakkitten: the version and description are stored in database. This description may be different from the description in script itself, and it is translated via Django gettext files (part of weechat.org site source code).

**Comment by oakkitten** [comment]
> This description may be different from the description in script itself

t-this looks like another problem...

anyways, start weechat, load script, get author/version/description/etc and update the database if necessary? don't see why it couldn't be done automatically

**Comment by flashcode** [comment]
The database is on weechat.org server and is used to generate the plugins.xml file (which is used by weechat to update the scripts list).
So a WeeChat client has no access to this database.
